src/System/Interactions.jl

```diff
@@ -325,53 +317,72 @@ function set_forces_aux!(B, dipoles::Array{Vec3, 4}, ints::Interactions, i::Int,
     end
 
     # Heisenberg exchange
-    for (; isculled, bond, J) in ints.heisen
-        isculled && break
-        for cellᵢ in cells
-            cellⱼ = offsetc(cellᵢ, bond.n, latsize)
-            sᵢ = dipoles[cellᵢ, bond.i]
-            sⱼ = dipoles[cellⱼ, bond.j]
-            B[cellᵢ, bond.i] -= J  * sⱼ
-            B[cellⱼ, bond.j] -= J' * sᵢ
-        end
+    foreachbond(ints.heisen) do J, site1, site2
+        sᵢ = dipoles[site1]
+        sⱼ = dipoles[site2]
+        B[site1] -= J  * sⱼ
+        B[site2] -= J' * sᵢ
     end
+
     # Quadratic exchange matrix
-    for (; isculled, bond, J) in ints.exchange
-        isculled && break
-        for cellᵢ in cells
-            cellⱼ = offsetc(cellᵢ, bond.n, latsize)
-            sᵢ = dipoles[cellᵢ, bond.i]
-            sⱼ = dipoles[cellⱼ, bond.j]
-            B[cellᵢ, bond.i] -= J  * sⱼ
-            B[cellⱼ, bond.j] -= J' * sᵢ
-        end
+    foreachbond(ints.exchange) do J, site1, site2
+        sᵢ = dipoles[site1]
+        sⱼ = dipoles[site2]
+        B[site1] -= J  * sⱼ
+        B[site2] -= J' * sᵢ
     end
+
     # Scalar biquadratic exchange
-    for (; isculled, bond, J) in ints.biquad
-        isculled && break
-        for cellᵢ in cells
-            cellⱼ = offsetc(cellᵢ, bond.n, latsize)
-            sᵢ = dipoles[cellᵢ, bond.i]
-            sⱼ = dipoles[cellⱼ, bond.j]
-
-            if sys.mode == :dipole
-                Sᵢ = (sys.Ns[cellᵢ, bond.i]-1)/2
-                Sⱼ = (sys.Ns[cellⱼ, bond.j]-1)/2
-                S = √(Sᵢ*Sⱼ)
-                # Renormalization introduces a factor r and a Heisenberg term
-                r = (1 - 1/S + 1/4S^2)
-                B[cellᵢ, bond.i] -= J * (2r*sⱼ*(sᵢ⋅sⱼ) - sⱼ/2)
-                B[cellⱼ, bond.j] -= J * (2r*sᵢ*(sᵢ⋅sⱼ) - sᵢ/2)
-            elseif sys.mode == :large_S
-                B[cellᵢ, bond.i] -= J * 2sⱼ*(sᵢ⋅sⱼ)
-                B[cellⱼ, bond.j] -= J * 2sᵢ*(sᵢ⋅sⱼ)
-            elseif sys.mode == :SUN
-                error("Biquadratic currently unsupported in SU(N) mode.")
-            end
+    foreachbond(ints.biquad) do J, site1, site2
+        sᵢ = dipoles[site1]
+        sⱼ = dipoles[site2]
+
+        if sys.mode == :dipole
+            Sᵢ = (sys.Ns[site1]-1)/2
+            Sⱼ = (sys.Ns[site2]-1)/2
+            S = √(Sᵢ*Sⱼ)
+            # Renormalization introduces a factor r and a Heisenberg term
+            r = (1 - 1/S + 1/4S^2)
+            B[site1] -= J * (2r*sⱼ*(sᵢ⋅sⱼ) - sⱼ/2)
+            B[site2] -= J * (2r*sᵢ*(sᵢ⋅sⱼ) - sᵢ/2)
+        elseif sys.mode == :large_S
+            B[site1] -= J * 2sⱼ*(sᵢ⋅sⱼ)
+            B[site2] -= J * 2sᵢ*(sᵢ⋅sⱼ)
+        elseif sys.mode == :SUN
+            error("Biquadratic currently unsupported in SU(N) mode.")
         end
     end
 end
 
+# Producer of a functions that iterate over all interactions for a given cell
+function inhomog_bond_iterator(latsize, cell)
+    return function inner(f, ints)
```

Comment:
Since the name isn't conveying new information, isn't it better to name it informatively,

```suggestion
    return function foreachbond(f, ints)
```

or not at all?

```suggestion
    return function (f, ints)
```